### ISSUE TITLE
Thanks for your code. Could you elaborate the implementation details of ViLT-CAP you used as one of the baselines?

### ISSUE THREAD

**Issue by meiling-fdu** [jacobswan1/ViTCAP#4]

**Comment by jacobswan1** [comment]
Great question. So when I adapt ViLT for captioning task, an important adaptions is to make the bidirectional attention mask to single-directional, which allows the auto-regressive prediction. Second, apply the pre-trained ViLT weight to the new architecture and trained ViLT on COCO with 15% (or different) language masking rate. The optimization objective is simply masked token classification loss. After training, you will find that the pre-training of ViLT is helping not quite obvious, which is mainly because it's pre-training on CC was conducted bidirectionally, hinders its performances to a very large extent.

**Comment by meiling-fdu** [comment]
Will it be convenient for you to publish the sorce code?

**Comment by jacobswan1** [comment]
It has been a while and I did not quite retain this part of code. But it should not be complex: adjust the attention map would do the job.
